lib/experimental/class-gutenberg-rest-global-styles-revisions-controller.php

```diff
@@ -0,0 +1,363 @@
+				),
+
+				// Adds custom global styles revisions schema.
+				'author_display_name' => array(
+					'description' => __( 'The display name of the author.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit' ),
+				),
+
+				'author_avatar_url'   => array(
+					'description' => __( 'A URL to the avatar image of the author', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit' ),
+				),
+
+				'date_display'        => array(
+					'description' => __( 'A human-friendly rendering of the date', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit' ),
+				),
+				// Adds settings and styles from the WP_REST_Global_Styles_Controller parent schema.
+				'styles'              => array(
+					'description' => __( 'Global styles.', 'gutenberg' ),
+					'type'        => array( 'object' ),
+					'context'     => array( 'view', 'edit' ),
+				),
+				'settings'            => array(
+					'description' => __( 'Global settings.', 'gutenberg' ),
+					'type'        => array( 'object' ),
+					'context'     => array( 'view', 'edit' ),
+				),
+			),
+		);
+
+		$this->schema = $schema;
+
+		return $this->add_additional_fields_schema( $this->schema );
+	}
+
+	/**
+	 * Checks if a given request has access to read a single global style.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+		$post = $this->get_parent( $request['parent'] );
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		if ( ! current_user_can( 'read_post', $post->ID ) ) {
+			return new WP_Error(
+				'rest_cannot_view',
+				__( 'Sorry, you are not allowed to view revisions for this global style.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get the parent post, if the ID is valid. Copied from WP_REST_Revisions_Controller.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param int $parent_post_id Supplied ID.
+	 * @return WP_Post|WP_Error Post object if ID is valid, WP_Error otherwise.
+	 */
+	protected function get_parent( $parent_post_id ) {
+		$error = new WP_Error(
+			'rest_post_invalid_parent',
+			__( 'Invalid post parent ID.', 'gutenberg' ),
+			array( 'status' => 404 )
+		);
+
+		if ( (int) $parent_post_id <= 0 ) {
+			return $error;
+		}
+
+		$parent_post = get_post( (int) $parent_post_id );
+
+		if ( empty( $parent_post ) || empty( $parent_post->ID )
+			|| $this->parent_post_type !== $parent_post->post_type
+		) {
+			return $error;
+		}
+
+		return $parent_post;
+	}
+
+	/**
+	 * Retrieves the query params for the global styles revision collection.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @return array Collection parameters.
+	 */
+	public function get_collection_params() {
```

Comment:
Again, I imagine this isn't required for the initial goal for this endpoint, but would we eventually include support for common collection args (like per page, offset, order, etc so that paginated responses would be possible?)

Reply:
Exactly, and, now that you've raised it, I can probably remove it from this iteration. I think it was a leftover from experimentation. Thank you!